pkg/clusterversion/clusterversion.go

```diff
@@ -90,3 +96,261 @@ func (builder *Builder) validate() (bool, error) {
+}
+
+// GetNextUpdateVersionImage fetches the next recommended or conditional update for the cluster.
+func (builder *Builder) GetNextUpdateVersionImage(stream string, acceptConditionalVersions bool) (string, error) {
+	if valid, err := builder.validate(); !valid {
+		return "", err
+	}
+
+	glog.V(100).Infof("Getting the update version image in stream %s for clusterversion %s",
+		stream, builder.Definition.Name)
+
+	if !builder.Exists() {
+		return "", fmt.Errorf("%s clusterversion not found", builder.Definition.Name)
+	}
+
+	if stream == "" {
+		return "", fmt.Errorf("stream can not be empty")
+	}
+
+	currentVersion := builder.Object.Status.Desired.Version
+
+	for _, availableUpdate := range builder.Object.Status.AvailableUpdates {
+		isStreamUpdate, err := builder.isStreamUpdate(currentVersion, availableUpdate.Version, stream)
+		if isStreamUpdate && err == nil {
+			return availableUpdate.Image, nil
+		}
+	}
+
+	if acceptConditionalVersions {
+		for _, conditionalUpdate := range builder.Object.Status.ConditionalUpdates {
+			isStreamUpdate, err := builder.isStreamUpdate(currentVersion, conditionalUpdate.Release.Version, stream)
+			if isStreamUpdate && err == nil {
+				return conditionalUpdate.Release.Image, nil
+			}
+		}
+	}
+
+	return "", fmt.Errorf("Update version in %s stream not found", stream)
+}
+
+// IsStreamUpdate checks if updateVersion is a 'stream' (X, Y or Z) update for version.
+func (builder *Builder) isStreamUpdate(version, updateVersion, stream string) (isStreamUpdate bool, err error) {
+	glog.V(100).Infof("Verify if updateVersion %s is a stream %s update for version %s", updateVersion, stream, version)
+
+	if !slices.Contains([]string{Y, Z, Y}, stream) {
+		glog.V(100).Infof("invalid stream %s", stream)
+
+		return false, fmt.Errorf("invalid stream %s", stream)
+	}
+
+	semVersion, semVersionError := semver.NewVersion(version)
+
+	if semVersionError != nil {
+		return false, fmt.Errorf("the version %s is invalid", version)
+	}
+
+	semUpdateVersion, updateVersionError := semver.NewVersion(updateVersion)
+	glog.V(100).Infof("Testing %s and %s", semVersion.String(), semUpdateVersion.String())
+
+	if updateVersionError != nil {
+		return false, fmt.Errorf("the Update Version %s is invalid", updateVersion)
+	}
+
+	switch major := semVersion.Major(); {
+	case major == semUpdateVersion.Major() && semVersion.Minor() == semUpdateVersion.Minor() &&
+		semVersion.Patch() < semUpdateVersion.Patch() && stream == Z:
+		glog.V(100).Infof("This version is a z update: %s", semUpdateVersion.String())
+
+		return true, nil
+
+	case major == semUpdateVersion.Major() &&
+		semVersion.Minor() < semUpdateVersion.Minor() && stream == Y:
+		glog.V(100).Infof("This version is a y update: %s", semUpdateVersion.String())
+
+		return true, nil
+
+	case semVersion.Major() < semUpdateVersion.Major() && stream == X:
+		glog.V(100).Infof("This version is an x update: %s", semUpdateVersion.String())
+
+		return true, nil
+
+	default:
+		return false, nil
```

Comment:
Please add glog message for this conditions. It'll improve visibility and debugging process later

Reply:
Added, thanks